packages/components/src/spinner/src/Spinner.tsx

```diff
@@ -0,0 +1,82 @@
+import { ComponentProps, forwardRef } from "react";
+import { isNil, InternalProps, OmitInternalProps, StyledComponentProps, cssModule, mergeProps, normalizeSize, createSizeAdapter } from "../../shared";
+import { Box } from "../../box";
+import { ResponsiveProp, useResponsiveValue } from "../../styling";
+import { Text } from "../../typography";
+
+const DefaultElement = "div";
+
+export interface InnerSpinnerProps extends InternalProps, StyledComponentProps<typeof DefaultElement> {
+    /**
+     * Specify whether you want the Spinner to be animating or not
+     */
+    active?: boolean;
+    /**
+     * A spinner can vary in size.
+     */
+    size?: ResponsiveProp<"sm" | "md" | "lg">;
+}
+
+/* eslint-disable sort-keys, sort-keys-fix/sort-keys-fix */
+const textSize = createSizeAdapter({
+    "sm": "sm",
+    "md": "md",
+    "lg": "lg"
+});
+/* eslint-enable sort-keys, sort-keys-fix/sort-keys-fix */
+
+export function InnerSpinner({
+    as = DefaultElement,
+    size = "md",
+    active = true,
+    forwardedRef,
+    children,
+    "aria-label": ariaLabel,
+    "aria-labelledby": ariaLabelledBy,
+    ...rest
+}: InnerSpinnerProps) {
+    const sizeValue = useResponsiveValue(size);
+
+    if (isNil(children) && isNil(ariaLabel) && isNil(ariaLabelledBy)) {
+        console.error("A spinner must either have children, an \"aria-label\" attribute or an \"aria-labelledby\" attribute.");
+    }
+
+    const labelMarkup = children && (
+        <Text className="o-ui-spinner-label" color="inherit" size={textSize(sizeValue)}>
+            {children}
+        </Text>
+    );
+
+    const spinnerMarkup = (
+        <div className="o-ui-spinner-wheel"></div>
+    );
+
+    return (
+        <Box
+            {...mergeProps(
+                rest,
+                {
+                    as,
+                    className: cssModule(
+                        "o-ui-spinner",
+                        normalizeSize(sizeValue),
+                        active && "active"
+                    ),
+                    ref: forwardedRef,
+                    role: "status"
```

Comment:
I am puzzled by this one, it seems like the right role might "progressbar" according to React Aria progresse bar indeterminate example: https://react-spectrum.adobe.com/react-spectrum/ProgressCircle.html#indeterminate.

It kind of seems to make sense after reading the following https://www.digitala11y.com/progressbar-role.

If progress role is the right role, there be additional step taken in some of Orbits components to correctly set the `aria-describedby` and `aria-busy` attribute. The Spinner components documentation should also be updated to mention those.

How did you come up with the "status" role?